### PR TITLE
Integration test script

### PR DESCRIPTION
**Pull request by amackillop**
Basic script for now to run CLI commands to end to end test the project. Requires having run `docker compose up` first. I was trying to get something more proper in Rust tests but was running into issues with docker compose exiting. This can do for now

### CODE REVIEW

File: run-integration-tests.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env bash
+set +eux pipefail
```
Comment (amlozano1) [inline run-integration-tests.sh:2]: Why do you need pipefail here? Not seeing any pipelines.

File: run-integration-tests.sh [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+# Usage: After running docker compose up, execute this script to check that each 
+# cli command succeeds. The CLI will make requests to the server.
+pushd cli
```
Comment (amlozano1) [inline run-integration-tests.sh:5]: Guessing this script only works from the root? 

I think you might want to remove this script or maybe put everything into `cargo test` or `just` or `make` file, whichever the team prefers

**Review by amlozano1 — APPROVED** [review]

File: run-integration-tests.sh (hunk 2 above)
Comment (amackillop) [reply run-integration-tests.sh:5]: I have another branch trying to do the proper cargo test route but it was taking too long so this is a stop gap to just run ourselves as a spot check.

Comment (amackillop) [reply run-integration-tests.sh:5]: Had issues with docker compose up exiting in the test prematurely

File: run-integration-tests.sh (hunk 1 above)
Comment (amackillop) [reply run-integration-tests.sh:2]: I guess we don't. Was a copilot auto complete

Comment (amlozano1) [reply run-integration-tests.sh:2]: I suspected as much, but didn't want to accuse you of committing AI code you didn't understand.

To be fair I asked the AI to explain what this line did in order to leave that comment.

**Review by saidalaoui-sq — APPROVED** [review]

**Pull request merged**